### DISCUSSION TITLE
Frequent pair requests, Google Pixel 6.

### DISCUSSION THREAD

**Discussion by reyalex1** [NightscoutFoundation/xDrip#2717] [Q&A]
Having frequent pair requests using Dexcom One, Google Pixel 6. Logs uploaded. Enabled detailed logs and one request should be visible today. This happened about 6 times yesterday without detailed logs. All settings are as per Navid's guide for Android 10 and above, including disabling of battery optimisation for Bluetooth.

**Comment by Navid200** [comment]
@jamorham

**Comment by Navid200** [comment]
It's always a good idea to include in a post asking for help the version of xDrip you are using.
In this case, it's also important to clarify if you have updated your phone to the latest available Android released by the manufacturer.

**Reply by reyalex1** [reply]
Hi Navid

My phone is Android 13. Please see attached showing xdrip version.
![Screenshot_20230304-164233~2](https://user-images.githubusercontent.com/85752598/222918284-39bc9323-8fed-4910-878c-8f15e25bd811.png)

**Reply by reyalex1** [reply]
This too, in case the security update date is off use.
![Screenshot_20230304-171600](https://user-images.githubusercontent.com/85752598/222919689-5085e2b2-7c2e-4e2c-afd3-2470cab35445.png)

**Reply by Navid200** [reply]
I'm not aware of any recent changes to the collector that would affect Pixel.
But, would you please update to the latest Nightly just to remove that possibility?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.02.26

**Reply by Navid200** [reply]
And also to ensure that engineering mode is disabled, would you please post a screenshot of the system status page?  You can blackout the transmitter ID.

**Reply by reyalex1** [reply]
Updated just now. I will report back if pair requests continue, or otherwise confirm they have stopped. 

Here is the system status page:
![Screenshot_20230304-193439](https://user-images.githubusercontent.com/85752598/222925447-a7b77765-6aca-428b-8970-03d73eef2c97.png)

**Reply by reyalex1** [reply]
Pair requests have continued after update.

**Reply by jamorham** [reply]
I have a couple of questions. How frequent are these pair requests, are they, for example every 20 minutes or does it work for hours and then you get pairing requests?

Can you also describe the history of the problem with this transmitter. Did it work without frequent pairing requests for a period of time and then stop working or has it had this issue since the moment you started using it?

**Reply by reyalex1** [reply]
It seems to be a few pair requests per day, so works for many hours in between. The transmitter has been connected since Thursday and it has been like it since then. I don't have any previous experience using Dexcom as this is the first transmitter, so cannot compare to a previous one.

Would it help if I upload additional logs?

**Reply by Navid200** [reply]
But, you used this same transmitter with an old phone and you did not have this problem with it.  Is that not true?

**Reply by reyalex1** [reply]
Yes I tested this transmitter with old phone (Samsung J5, Android 9) before I was able to connect successfully with the Pixel 6. I left that connected only for a few hours before attempting again with the Pixel, from that time period I couldn't say for sure that the issue did not exist. That phone has been switched off since.

**Reply by Navid200** [reply]
Just for clarification, do you mind posting a screenshot of the following page?
Settings &#8722;> G5/G6/Dex1 Debug Settings

**Reply by reyalex1** [reply]
![Screenshot_20230305-153859](https://user-images.githubusercontent.com/85752598/222970409-6c7c7ba0-fe02-4d4c-9fde-bb11d0f46fce.png)
![Screenshot_20230305-153912](https://user-images.githubusercontent.com/85752598/222970423-571fbd86-4ba4-44ac-94c1-828b27981061.png)

**Reply by Navid200** [reply]
I use G6.  When there is a connectivity issue, the watchdog turns Bluetooth off and back on.  But, it never unpairs.
But, I have never had a Dexcom One.
This may be specific to the new type of transmitter you have.
Let's see what @jamorham thinks.

**Reply by Navid200** [reply]
Go to Android Settings &#8722;> Apps & notifications &#8722;> xDrip+ &#8722;> Permissions &#8722;> Location

What do you see?

**Reply by reyalex1** [reply]
Sightly different menu options to get there but these are the location settings for xdrip:

![Screenshot_20230305-204536](https://user-images.githubusercontent.com/85752598/222985022-1642cf6e-ef3e-4e4a-91e0-1be4dff82d33.png)

**Comment by jamorham** [comment]
Thanks for providing additional information. Could you also enable OB1 debugging as shown here: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector and then after you get a pairing request and complete that and get a few readings could you upload logs again. Feel free to upload as many logs as you want over different periods of time and I will review those.

**Reply by reyalex1** [reply]
@jamorham thanks. I'll upload some logs now then enable and upload again after the next request. Thankfully no requests for about 18 hours give or take.

**Reply by reyalex1** [reply]
@jamorham I've just realised these are the same settings Navid asked me to enable previously (outside of this discussion). So you should already have detailed info from the logs uploaded initially and just now.
